extension/landing-zone/variables.tf

```diff
@@ -0,0 +1,328 @@
+    public_key = optional(string)
+  })
+
+  validation {
+    error_message = "Public SSH Key must be a valid ssh rsa public key."
+    condition     = var.ssh_keys.public_key == null || can(regex("ssh-rsa AAAA[0-9A-Za-z+/]+[=]{0,3} ?([^@]+@[^@]+)?", var.ssh_keys.public_key))
+  }
+}
+
+variable "resource_tags" {
+  description = "List of tags for the VSI, Block Storage, Security Group, Floating IP and Load Balancer created"
+  type        = list(string)
+  default     = null
+}
+
+variable "access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the VSI resources created by the module."
+  default     = []
+}
+
+variable "image_name" {
+  description = "Image ID used for VSI. Run 'ibmcloud is images' to find available images. Be aware that region is important for the image since the id's are different in each region."
+  type        = string
+  default     = "ibm-ubuntu-22-04-2-minimal-amd64-1"
+}
+
+variable "machine_type" {
+  description = "VSI machine type"
+  type        = string
+  default     = "cx2-2x4"
+}
+
+variable "security_group" {
+  description = "Security group created for the VSI"
+  type = object({
+    name = string
+    rules = list(
+      object({
+        name      = string
+        direction = string
+        source    = string
+        tcp = optional(
+          object({
+            port_max = number
+            port_min = number
+          })
+        )
+        udp = optional(
+          object({
+            port_max = number
+            port_min = number
+          })
+        )
+        icmp = optional(
+          object({
+            type = number
+            code = number
+          })
+        )
+      })
+    )
+  })
+  default = {
+    "name" : "management",
+    "rules" : [
+      {
+        "direction" : "inbound",
+        "name" : "allow-ibm-inbound",
+        "source" : "161.26.0.0/16"
+      },
+      {
+        "direction" : "inbound",
+        "name" : "allow-vpc-inbound",
+        "source" : "10.0.0.0/8"
+      },
+      {
+        "direction" : "outbound",
+        "name" : "allow-vpc-outbound",
+        "source" : "10.0.0.0/8"
+      },
+      {
+        "direction" : "outbound",
+        "name" : "allow-ibm-tcp-53-outbound",
+        "source" : "161.26.0.0/16",
+        "tcp" : {
+          "port_max" : 53,
+          "port_min" : 53
+        }
+      },
+      {
+        "direction" : "outbound",
+        "name" : "allow-ibm-tcp-80-outbound",
+        "source" : "161.26.0.0/16",
+        "tcp" : {
+          "port_max" : 80,
+          "port_min" : 80
+        }
+      },
+      {
+        "direction" : "outbound",
+        "name" : "allow-ibm-tcp-443-outbound",
+        "source" : "161.26.0.0/16",
+        "tcp" : {
+          "port_max" : 443,
+          "port_min" : 443
+        }
+      }
+    ]
+  }
+}
+
+variable "user_data" {
+  description = "User data to initialize VSI deployment"
+  type        = string
+  default     = null
+}
+
```

Comment:
`placement_group_id` needs to be exposed now that https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone-vsi/pull/539 has been merged